### PR TITLE
only write required properties with no default and new default value

### PR DESCRIPTION
**Pull request by Emptyless**
when `minimal: true` is set, only write required properties that have no default value configured. 

Also, the default value is changed to `null` instead of the default value for a type

### CODE REVIEW

File: config_test.go [hunk 1]
```diff
@@ -43,6 +43,18 @@ func TestConfig_ForProperty_ReturnsExpectedConfig(t *testing.T) {
 				ValueOverrides: map[string]any{"bar": "baz"},
 			},
 		},
+		"subproperty is returned with minimal=true if set on parent": {
+			input: &Config{
+				Minimal:        true,
+				ValueOverrides: map[string]any{"foo": map[string]any{"bar": "baz"}},
```
Comment (survivorbat) [inline config_test.go]: I'd remove the ValueOverrides here, as they're not relevant for the test afaik

File: schema.go [hunk 2]
```diff
@@ -101,13 +123,9 @@ func (j *JSONSchema) ScheYAML(cfg *Config) *yaml.Node {
 		case j.Default != nil:
 			result.Value = fmt.Sprint(j.Default)
 
-		case j.Type == TypeString:
-			result.LineComment = cfg.TODOComment
-			result.SetString("")
-
 		default:
 			result.LineComment = cfg.TODOComment
-			result.Value = j.Type.DefaultValue()
```
Comment (survivorbat) [inline schema.go:110]: In that case, that method can be removed entirely :D

File: schema_test.go [hunk 3]
```diff
@@ -96,7 +125,7 @@ func TestJSONSchemaObject_JSON_PreservesExtraProperties(t *testing.T) {
             "name": {
               "type": "string", 
               "description": "The name of the beverage", 
-              "examples": ["Coffee", "Tea", "Cappucino"]
+              "examples": ["Coffee", "Tea", "Cappuccino"]
```
Comment (survivorbat) [inline schema_test.go:128]: oops

File: config.go [hunk 4]
```diff
@@ -24,6 +24,9 @@ type Config struct {
 	// TODOComment is used in case no default value was defined for a property. It is set by
 	// default in NewConfig but can be emptied to remove the comment altogether.
 	TODOComment string
+
+	// Minimal mode only renders properties which are required and have no default value
+	Minimal bool
```
Comment (survivorbat) [inline config.go:29]: Could you add a `Minimal()` (or whatever you think makes sense) as a functional option

**Review by survivorbat — CHANGES_REQUESTED** [review]

File: config_test.go (hunk 1 above)
Comment (Emptyless) [reply config_test.go]: Removed

File: schema.go (hunk 2 above)
Comment (Emptyless) [reply schema.go:110]: Removed

File: config.go (hunk 4 above)
Comment (Emptyless) [reply config.go:29]: Added

**Review by survivorbat — APPROVED** [review]
Nice addition!

**Pull request merged**